biogeochem/EDCohortDynamicsMod.F90

```diff
@@ -736,52 +727,14 @@ subroutine terminate_cohorts( currentSite, currentPatch, level )
                    currentCohort%n * (struct_c+sapw_c+leaf_c+fnrt_c+store_c+repro_c)
           end if
 
-          !put the litter from the terminated cohorts straight into the fragmenting pools
-          if (currentCohort%n.gt.0.0_r8) then
-             do c=1,ncwd
-
-                currentPatch%CWD_AG(c)  = currentPatch%CWD_AG(c) + currentCohort%n*(struct_c+sapw_c) / &
-                     currentPatch%area &
-                     * SF_val_CWD_frac(c) * EDPftvarcon_inst%allom_agb_frac(currentCohort%pft) 
-                currentPatch%CWD_BG(c)  = currentPatch%CWD_BG(c) + currentCohort%n*(struct_c+sapw_c) / &
-                     currentPatch%area &
-                     * SF_val_CWD_frac(c) * (1.0_r8 -  EDPftvarcon_inst%allom_agb_frac(currentCohort%pft)) 
-             enddo
-             
-             currentPatch%leaf_litter(currentCohort%pft) = currentPatch%leaf_litter(currentCohort%pft) + currentCohort%n* &
-                  (leaf_c)/currentPatch%area
-
-             currentPatch%root_litter(currentCohort%pft) = currentPatch%root_litter(currentCohort%pft) + currentCohort%n* &
-                  (fnrt_c+store_c)/currentPatch%area 
-
-
-             ! keep track of the above fluxes at the site level as a CWD/litter input flux (in kg / site-m2 / yr)
-             do c=1,ncwd
-                currentSite%CWD_AG_diagnostic_input_carbonflux(c)  = currentSite%CWD_AG_diagnostic_input_carbonflux(c) &
-                     + currentCohort%n*(struct_c + sapw_c) * &
-                     SF_val_CWD_frac(c) * EDPftvarcon_inst%allom_agb_frac(currentCohort%pft) * hlm_days_per_year / AREA
-                currentSite%CWD_BG_diagnostic_input_carbonflux(c)  = currentSite%CWD_BG_diagnostic_input_carbonflux(c) &
-                     + currentCohort%n*(struct_c + sapw_c) * &
-                     SF_val_CWD_frac(c) * (1.0_r8 -  EDPftvarcon_inst%allom_agb_frac(currentCohort%pft))  * hlm_days_per_year / AREA
-             enddo
-             
-             currentSite%leaf_litter_diagnostic_input_carbonflux(currentCohort%pft) = &
-                  currentSite%leaf_litter_diagnostic_input_carbonflux(currentCohort%pft) +  &
-                  currentCohort%n * (leaf_c) * hlm_days_per_year  / AREA
-             currentSite%root_litter_diagnostic_input_carbonflux(currentCohort%pft) = &
-                  currentSite%root_litter_diagnostic_input_carbonflux(currentCohort%pft) + &
-                  currentCohort%n * (fnrt_c + store_c) * hlm_days_per_year  / AREA
+          ! put the litter from the terminated cohorts 
+          ! straight into the fragmenting pools
 
+          if (currentCohort%n.gt.0.0_r8) then
+             call SendCohortToLitter(currentSite,currentPatch, &
+                  currentCohort,currentCohort%n)
           end if
           
-          ! Zero out the state pools
```

Comment:
Good question.  This is because this routine is meant to send the whole plant to the litter pool. The SetState() routines adjust the per-individual mass pools for each plant organ.  Since we are sending whole plants to the litter pool,  we don't need to modify the component pools of the plant, we simply need to just reduce the number density.